src/plugins/lvm-dbus.c

```diff
@@ -2197,7 +2213,7 @@ static gboolean filter_lvs_by_vg (const gchar **lvs, const gchar *vg_name, GSLis
             }
 
             if (g_strcmp0 (lv_vg_name, vg_name) == 0) {
-                *out = g_slist_prepend (*out, (gchar *) *lv_p);
+                *out = g_slist_prepend (*out, g_strdup((gchar *) *lv_p));
```

Comment:
This function should avoid creating duplicates.

Reply:
Given how this function is used, it's difficult to not duplicate the string in a clean way. It would either have to deallocate strings that don't match the VG filter internally and then unref just the variant container afterwards, or g_strfreev() would have to be replaced with loops that free the strings in the input vector that are not in the matched_lvs vector.

```
    lvs = get_existing_objects (LV_OBJ_PREFIX, error);
    if (!lvs && (*error)) {
        /* error is already populated */
        return NULL;
    }
    success = filter_lvs_by_vg (lvs, vg_name, &matched_lvs, &n_lvs, error);
    g_strfreev ((gchar **) lvs);
    if (!success)
        return NULL;
```

Reply:
I'd suggest to make this duplicate-records concern separate from this pull request.

Reply:
> Given how this function is used, it's difficult to not duplicate the string in a clean way. It would either have to deallocate strings that don't match the VG filter internally

Sorry, but I don't follow. Isn't the original code in the function doing exactly that? The function takes a pointer to an array of strings, the matching ones are moved elsewhere, the non-matching ones are free'd. And then this array itself is free'd after the function call. Don't really see a problem here. What am I missing?

Reply:
After looking at that function again, I see what you're saying. I'll clean it up and test with valgrind tomorrow.